exp/controllers/rosamachinepool_controller.go

```diff
@@ -357,14 +357,12 @@ func (r *ROSAMachinePoolReconciler) reconcileMachinePoolVersion(machinePoolScope
 }
 
 func (r *ROSAMachinePoolReconciler) updateNodePool(machinePoolScope *scope.RosaMachinePoolScope, ocmClient *ocm.Client, nodePool *cmv1.NodePool) (*cmv1.NodePool, error) {
-	desiredSpec := *machinePoolScope.RosaMachinePool.Spec.DeepCopy()
-	currentSpec := nodePoolToRosaMachinePoolSpec(nodePool)
+	machinePool := machinePoolScope.RosaMachinePool.DeepCopy()
+	// default all fields before comparing, so that nil/unset fields don't cause an unnecessary update call.
+	machinePool.Default()
```

Comment:
shouldn't this have been defaulted by the webhook when we get here?

Reply:
yes it should, added here incase the use decided to not run the webhook.